pkg/runtime/embedded.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/conversion"
+)
+
+// NewSerializedConversions handles the conversion between Object and RawExtension given
+// a supplied serializer. If conversion is desired, the caller provide a serializer
+// that performs opinionated conversions.
+func NewSerializedConversions(serializer Serializer) *conversion.ConversionFuncs {
+	fns := conversion.NewConversionFuncs()
+	fns.Register(
+		func(in Object, out *RawExtension, s conversion.Scope) error {
+			if in == nil {
+				out.RawJSON = []byte("null")
+				return nil
+			}
+			data, err := Encode(serializer, in)
+			if err != nil {
+				return err
+			}
+			out.RawJSON = data
```

Comment:
Yes, glad you agree.  Mind reading


On Dec 1, 2015, at 7:26 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/runtime/embedded.go
<https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46361738>:

> +// NewSerializedConversions handles the conversion between Object and RawExtension given
> +// a supplied serializer. If conversion is desired, the caller provide a serializer
> +// that performs opinionated conversions.
> +func NewSerializedConversions(serializer Serializer) *conversion.ConversionFuncs {
> +	fns := conversion.NewConversionFuncs()
> +	fns.Register(
> +		func(in Object, out *RawExtension, s conversion.Scope) error {
> +			if in == nil {
> +				out.RawJSON = []byte("null")
> +				return nil
> +			}
> +			data, err := Encode(serializer, in)
> +			if err != nil {
> +				return err
> +			}
> +			out.RawJSON = data

Perhaps change the name of this field to just "Raw" and also add a field
indicating the format of the data?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17922/files#r46361738>.